### DISCUSSION TITLE
Inspecting logits, and disabling prefill

### DISCUSSION THREAD

**Discussion by awgr** [ggml-org/llama.cpp#9725] [Q&A]
Hi, some questions:
(1) is there a straightforward way to dump logits for a new model being added to ggml? I read through the eval callback and the output building, but it's a bit opaque to me at what point the logits are actually materialized into the lctx.logits array.

(2) is there a way to constrain the input tokens per forward to 1? It would help if the input shapes are relatively static. I noticed ubatch appears to change the sequence length for the first forward pass, but it's not clear to me whether that behaves as I'd want it to (I more or less don't need batching at the moment)

(3) I'd like to disable most/all samplers, but I noticed that's a bit tricky. It seems to always include...
sampler chain: logits -> logit-bias -> penalties -> softmax -> dist

Thanks!
